### PR TITLE
Conditional stylecop activation

### PR DESCRIPTION
**Pull request by VPeruS**
### Purpose of this PR
Currently full build process eat 15 min, without stylecop rules it takes only 2 min, rules should be executed only on contribution to upstream.

### Testing status
Not tested.

### Comments
Open this pull request for discussion.

### CODE REVIEW

**Comment by Nihlus** [comment]
This is a good idea. StyleCop defaults to running if the <RunCodeAnalysis> tag is not present, however, so you'll have to take that into account - as you can see, both AppVeyor and Travis fail now.

**Comment by daerogami** [comment]
Are we documenting this in a central location? I'm worried about there being quirks that contributors have to be concerned with that only maintainers are aware of. Perhaps add something to the wiki (like a Build Guide for Contributors) for details like this?

**Comment by VPeruS** [comment]
They will get CI error. But adding link in readme to "Contributors guide"(and creating it) will be nice

**Comment by VPeruS** [comment]
I edited wiki page(why I have permission to do this?)

File: appveyor.yml [hunk 1]
```diff
@@ -2,7 +2,8 @@ image: Visual Studio 2017
 init:
   - git config --global core.autocrlf input
 build_script:
-  - cmd: build.cmd NuGet /p:BuildWithStyleCop=yes
```
Comment (daerogami) [inline appveyor.yml]: I personally would squash your last three commits into one. Otherwise I don't see any issues.

Comment (VPeruS) [reply appveyor.yml]: Seems appveyor didn't exec build process right, as it take 5 mins to build vs travis 20 min

**Comment by VPeruS** [comment]
Closed to not dublicate duplicate

**Pull request closed without merging**